arcgis-ios-sdk-samples/Maps/Create and save a map/SaveAsViewController.swift

```diff
@@ -38,12 +45,51 @@ class SaveAsViewController: UITableViewController {
             presentAlert(message: "Please enter a title.")
             return
         }
-        
+        // Get the tags from the text field.
         let tags = tagsTextField.text?
             .components(separatedBy: ",")
             .map { $0.trimmingCharacters(in: .whitespacesAndNewlines) } ?? []
-        
+        // Get the item description from the text field.
         let itemDescription = descriptionTextField.text ?? ""
-        delegate?.saveAsViewController(self, didInitiateSaveWithTitle: title, tags: tags, itemDescription: itemDescription)
+        
+        delegate?.saveAsViewController(self, didInitiateSaveWithTitle: title, tags: tags, itemDescription: itemDescription, folder: selectedFolder)
+    }
+    
+    /// Present the list of folders to choose from.
+    func showFolderOptions() {
+        // Set index for the default option.
+        let selectedIndex = portalFolders.firstIndex { $0 == selectedFolder } ?? portalFolders.count
+        // The titles of the portal folders and a "No folder" option.
+        let folderTitles = portalFolders.map { $0.title! } + ["No folder"]
+        // Prepare the options table view controller and handle selection.
+        let optionsViewController = OptionsTableViewController(labels: folderTitles, selectedIndex: selectedIndex) { [weak self] newIndex in
+            guard let self = self else { return }
+            // If "No folder" was selected, set selectedFolder to nil.
+            if newIndex == self.portalFolders.count {
+                self.selectedFolder = nil
+            } else {
+                // Select the appropriate folder.
+                self.selectedFolder = self.portalFolders[newIndex]
+            }
+            // Replace the label text with the selected option.
+            self.folderLabel.text = self.selectedFolder?.title ?? "No folder"
+            self.navigationController?.popViewController(animated: true)
+        }
+        optionsViewController.title = "Folders"
+        show(optionsViewController, sender: self)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+}
+
+extension SaveAsViewController /* UITableViewDelegate */ {
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let folderCell = IndexPath(row: 3, section: 0)
+        if indexPath == folderCell {
+            tableView.deselectRow(at: folderCell, animated: true)
```

Comment:
This is not needed (see `UITableViewController.clearsSelectionOnViewWillAppear`) and in fact prevents the neat deselection effect seen when interactively swiping from the left to pop the options view controller.

```suggestion
```